### PR TITLE
[JENKINS-42955] Add support for folder credentials

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
[JENKINS-42955](https://issues.jenkins-ci.org/browse/JENKINS-42955)

With this change the step uses the job as context instead of the jenkins instance.

There is a new JUnit test and also I have manually tested that a folde rcredential can be used.

@reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubNotificationPipelineStepTest.java [hunk 1]
```diff
@@ -291,6 +296,36 @@ public void build() throws Exception {
...
+        CredentialsStore folderStore = getFolderStore(f);
+        folderStore.addCredentials(Domain.global(),
+                new DummyCredentials(CredentialsScope.GLOBAL, "user", "password"));
+
+        WorkflowJob p = f.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "githubNotify account: 'raul-arabaolaza', context: 'PCT Results', " +
+                        "credentialsId: 'dummy', description: 'PCT Is OK', " +
+                        "repo: 'acceptance-test-harness', sha: '0b5936eb903d439ac0c0bf84940d73128d5e9487', " +
+                        "status: 'SUCCESS', targetUrl: 'http://www.cloudbees.com'"
```
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubNotificationPipelineStepTest.java]: `PCT is OK`but the repo is `acceptance-test-harness`? 😄

File: src/test/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubNotificationPipelineStepTest.java [hunk 2]
```diff
@@ -291,6 +296,36 @@ public void build() throws Exception {
     }
 
     @Test
+    public void buildWitFolderCredentials() throws Exception {
```
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/pipeline/githubstatusnotification/GitHubNotificationPipelineStepTest.java]: typo

**Comment by Dohbedoh** [comment]
LGTM, not sure why the build is interrupted though

**Comment by raul-arabaolaza** [comment]
@reviewbybees Done

**Pull request merged**